doxyfile_options

```diff
@@ -32,7 +32,7 @@ DOXYFILE_ENCODING      = UTF-8
 # title of most generated pages and in a few other places.
 # The default value is: My Project.
 
-PROJECT_NAME           = "My Project"
+PROJECT_NAME           = "Mbed OS Reference"
```

Comment:
This change is also needed in the json version

Reply:
Can you please point me to which json file I should add this change? Then the group changes are only to make navigation through the API docs easier. It doesn't change or move any APIs or the groups. See the attached image. It shows how the grouping/API looks like with these changes. Earlier everything was showing up in the index.html page itself and this change cleans up lot of those. Its based on our discussion to align things like how HAL group was organized.
![api_groups](https://user-images.githubusercontent.com/31017510/32059421-b477133c-ba31-11e7-9253-f1b1db10bb01.PNG)

Reply:
doxyfile_options.json

Reply:
@sg- The contents of json file is as below - 
{
    "ENABLE_PREPROCESSING": "YES",
    "MACRO_EXPANSION": "YES",
    "EXPAND_ONLY_PREDEF": "NO",
    "SEARCH_INCLUDES": "YES",
    "INCLUDE_PATH": "", 
    "INCLUDE_FILE_PATTERNS": "", 
    "PREDEFINED": "DOXYGEN_ONLY DEVICE_ANALOGIN DEVICE_ANALOGOUT DEVICE_CAN DEVICE_ETHERNET DEVICE_EMAC DEVICE_FLASH  DEVICE_I2C DEVICE_I2CSLAVE DEVICE_I2C_ASYNCH DEVICE_INTERRUPTIN DEVICE_LOWPOWERTIMER DEVICE_PORTIN DEVICE_PORTINOUT DEVICE_PORTOUT DEVICE_PWMOUT DEVICE_RTC DEVICE_TRNG DEVICE_SERIAL DEVICE_SERIAL_ASYNCH DEVICE_SERIAL_FC DEVICE_SLEEP DEVICE_SPI DEVICE_SPI_ASYNCH DEVICE_SPISLAVE DEVICE_STORAGE \"MBED_DEPRECATED_SINCE(f, g)=\" \"MBED_ENABLE_IF_CALLBACK_COMPATIBLE(F, M)=\"",
    "EXPAND_AS_DEFINED": "", 
    "SKIP_FUNCTION_MACROS": "NO",
    "EXCLUDE_PATTERNS": "*/tools/* */targets/* */FEATURE_*/* */features/mbedtls/* */features/storage/* */features/unsupported/* */features/filesystem/* */BUILD/* */rtos/TARGET_CORTEX/rtx*/* */cmsis/* */features/FEATURES_*"
}
Looks like not changes are needed in that, can you please confirm?

Reply:
If you dont add that here, it will not be present in the online hosted version. All changes to the local doxyfile need to be accounted for in the json version. Otherwise it is assumed that the tool defaults will be used.

Reply:
Ah yes, I see your point now, updated json as well now. Please review.